torch/csrc/jit/ivalue.h

```diff
@@ -125,6 +126,22 @@ struct TORCH_API IValue final {
     return at::Tensor(toIntrusivePtr<at::TensorImpl, at::UndefinedTensorImpl>());
   }
 
+  IValue(caffe2::Blob blob)
+  : tag(Tag::Blob), is_intrusive_ptr(true) {
+    // TODO (after Tensor merge) If we pass in a Blob holding a Tensor, extract and
```

Comment:
rather, after tensor merge if we indeed use IValue in the executor - we should forbid Blob holding tensor all together and have special tensor-focused impl go directly to IValue. Thus Blob would become "everything else" implementation detail

Reply:
I agree with that in the long term, but in the short term it'll be an easier refactor if we just go through caffe2 and wrap Blob in IValue everywhere.